sklearn/neighbors/tests/test_neighbors.py

```diff
@@ -658,6 +659,15 @@ def test_radius_neighbors_regressor(n_samples=40,
             y_pred = neigh.predict(X[:n_test_pts] + epsilon)
             assert_true(np.all(abs(y_pred - y_target) < radius / 2))
 
+            # test fix for issue #9654
+            # test that nan is returned when no nearby observations
+            X_test_nan = np.ones((1, n_features))*-1
+            if weights == 'uniform':
+                assert_warns_message(RuntimeWarning, "Mean of empty slice.",
```

Comment:
No, we should issue a warning in all casess, not dependent on weights.

Reply:
You should also get the output, i.e. `pred = assert_warns_message(...)` so that you can then do the isnan assertion.

Reply:
Hi again, thanks for the feedback! Just to be clear on what you want me to do: raise a warning for RadiusNeighborRegressor when it returns one or more nan values irrespective of using weights or not. I guess this should be consistent with and without weights. So should we copy the warning from np.mean (without weights) and issue it when y_pred contains nan rows (and there are weights)?

Reply:
I suppose so. Either that, or we can be kinder to the user: hide the numpy error, and raise our own which is more explicit, with a message like "Some samples have no neighbors; predicting NaN."